### PR TITLE
Implement `Option::replace` in the core library

### PR DESCRIPTION
**Pull request by Kerollmops**
Here is the implementation of the `Option::replace` method. The first step of [the tracking issue #51998](https://github.com/rust-lang/rust/issues/51998).

### CODE REVIEW

File: src/libcore/lib.rs [hunk 1]
```diff
@@ -122,6 +122,7 @@
 #![feature(const_slice_len)]
 #![feature(const_str_as_bytes)]
 #![feature(const_str_len)]
+#![feature(option_replace)]
```
Comment (Kerollmops) [inline src/libcore/lib.rs]: Is it mandatory ?

**Comment by Kerollmops** [comment]
@Kimundi could you please give me feedbacks ? :smiley:

**Comment by pietroalbini** [comment]
Ping from triage @Kimundi! This PR needs your review.

**Comment by Kimundi** [comment]
Alright, looks good so far! Could you add some basic unit tests for the method as well? Basically just something that replicates the doc example.

File: src/libcore/option.rs [hunk 2]
```diff
@@ -845,6 +845,33 @@ impl<T> Option<T> {
     pub fn take(&mut self) -> Option<T> {
         mem::replace(self, None)
     }
+
+    /// Replaces the actual value in the option by the value given in parameter,
+    /// returning the old value if present,
+    /// leaving a `Some` in its place without deinitializing either one.
```
Comment (Kerollmops) [inline src/libcore/option.rs]: Ooops.. And I forgot to add `[]` around `Some`.

**Pull request merged**